addons/account_asset/account_asset_view.xml

```diff
@@ -61,16 +74,222 @@
         <field name="model">account.asset.category</field>
         <field name="arch" type="xml">
             <search string="Search Asset Category">
-                <field name="name" string="Asset Category"/>
+                <filter icon="terp-check" string="Sales" domain="[('type','=', 'sales')]" help="Revenue Recognitions"/>
+                <filter icon="terp-dialog-close" string="Purchase" domain="[('type','=', 'purchase')]" help="Assets"/>
+                <field name="name" string="Category"/>
                 <field name="journal_id"/>
+                <group expand="0" string="Group By...">
+                    <filter string="Type" domain="[]" context="{'group_by':'type'}"/>
+                </group>
+            </search>
+        </field>
+    </record>
+    
+    <record id="view_account_move_line_filter_new" model="ir.ui.view">
```

Comment:
Why do you create a new search view instead of inheriting to add the missing field? I would remove this view completly (there is a way to add any filter on any field not, so it's not required anymore)